### PR TITLE
Convert various tests to pytest style

### PR DESCRIPTION
**Pull request by radarhere**
Helps #4193

### CODE REVIEW

File: Tests/test_imageshow.py [hunk 1]
```diff
@@ -1,57 +1,61 @@
...
 
-        viewer = TestViewer()
-        ImageShow.register(viewer, -1)
 
-        for mode in ("1", "I;16", "LA", "RGB", "RGBA"):
-            with hopper() as im:
-                self.assertTrue(ImageShow.show(im))
-            self.assertTrue(viewer.methodCalled)
+@pytest.mark.skipif(
+    not on_ci() or not (is_win32() and on_github_actions()),
```
Comment (hugovk) [inline Tests/test_imageshow.py]: This was:
```python
    @unittest.skipUnless(
        on_ci() and not (is_win32() and on_github_actions()),
        "Only run on CIs; hangs on Windows on GitHub Actions",
    )
```

So inverting for `skipif`:

```suggestion
    not on_ci() or (is_win32() and on_github_actions()),
```

**Comment by wiredfool** [comment]
Something recently has kicked codecov into posting/sending emails. I'd really prefer to have it be a bit quieter about it's job.

**Comment by hugovk** [comment]
Apparently a change on Codecov's side, it should be quieter again when this propogrates around the branches: https://github.com/python-pillow/Pillow/pull/4386.

**Comment by wiredfool** [comment]
Ok, thanks.

**Comment by radarhere** [comment]
@hugovk that PR was actually already included in master before I forked this one

File: Tests/test_imageshow.py (hunk 1 above)
Comment (radarhere) [reply Tests/test_imageshow.py]: Thanks

**Comment by hugovk** [comment]
OK, let's look into that again. Created https://github.com/python-pillow/Pillow/issues/4394 to track it.

**Pull request merged**